source/Engine.cpp

```diff
@@ -951,6 +955,25 @@ void Engine::Draw() const
 			RingShader::Draw(pos, radius, 1.5f, it.disabled, color[6 + it.type], dashes, it.angle);
 	}
 
+	Point dCenter = player.FlagshipPtr()->Position();
+	for(const shared_ptr<Ship> &ship : ships)
```

Comment:
Each each ship to draw is drawn in `AddSprites`. Why not move the loop body there, since that's where each ship is drawn. No need to loop through every ship again.

Reply:
Because the effects makes use of the outline shader, it gets drawn directly when it's called. `AddSprites`, however, places all the sprites into a vector that is later drawn in the `Engine::Draw()` function. `Engine::Draw()` is also where the outline shader is right now, after all the sprites in the vector are drawn. If the outline shader goes into `AddSprites()`, it will later be drawn over by the ship's sprite, and have no effect.

TL;DR:
The outline shader only works after the rest of the sprites are drawn under it, which is in the`Draw()` function. Moving it to `AddSprites` will break it, as `AddSprites` is not where the sprites are drawn.